plugin-build/src/main/kotlin/io/sentry/android/gradle/util/SentryAndroidSdkState.kt

```diff
@@ -0,0 +1,32 @@
+package io.sentry.android.gradle.util
+
+import java.io.Serializable
+
+enum class SentryAndroidSdkState(val minVersion: String) : Serializable {
+    MISSING(""),
+
+    PERFORMANCE("4.0.0"),
+
+    FILE_IO("5.5.0");
+
+    fun isAtLeast(state: SentryAndroidSdkState): Boolean = this.ordinal >= state.ordinal
+
+    companion object {
+        /* ktlint-disable max-line-length */
+        val semverRegex =
+            Regex(
+                "((([0-9]+)\\.([0-9]+)\\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?)"
+            )
+        /* ktlint-enable max-line-length */
+
+        fun from(semVer: String): SentryAndroidSdkState =
+            when {
+                semVer == "unspecified" -> MISSING
+                !semverRegex.matches(semVer) -> error("Unknown version $semVer of sentry-android")
+                semVer < PERFORMANCE.minVersion -> MISSING
```

Comment:
It'll require a good level of attention to add new ones here. Maybe worth a comment?

Reply:
sure, you mean a comment mentioning that if we add a new instrumentation, we should also make sure this class is updated, right?